android-sdk/README.md

```diff
@@ -0,0 +1,224 @@
+
+## Proposed solution
+
+### Core SDK
+
+The Core SDK will be the package that all service modules inherit from.  It will be responsible for managing network requests, threading, logging, error handling and resolution, loading configuration for mobile-core services, and other cross cutting concerns.  Modules will interface with the core SDK by implementing the necessary interfaces provided and annotating the implementations so that the Core SDK will be able to correctly manage them as well.  The Core SDK will also allow some of its internals to be exposed through interfaces for purposes such as testing, bespoke integrations, and management of additional resources.
+
+The Core SDK will provide a request lifecycle that the Psuedocode section will demonstrate.
+
+### Service Modules
+
+Service modules will each represent a singular service that is supportd by mobile-core.  We will use, where available and practical, existing libraries for services and in those cases the modules will provide the necessary implementation of Core APIs to make them compatable with the Core SDK lifecycles and features.  The first Services we are looking at integrations will be Keycloak, fh-sync-service, and a RESTful API service module.
+
+### Implementation notes
+
+Public APIs should always expose native Java Objects and not stringified JSON or JSON Objects.
+
+### Use cases and psuedo code examples
+
+#### Starting up Mobile Core
+MobileCore is the "main" class for the SDK of the Core SDK.  Users instanciate it by placing a client configuration file in their assets directory, configuring MobileCore.Builder, and then calling its build method.
+
+```
+MobileCore.Builder builder = new MobileCore.Builder(application);
+MobileCore core = builder.build();
+```
+
+During the build method, MobileCore will scan a registry of services and configure them by calling the method `bootstrap` defined in the ServiceModule interface.  MobileCore will pass to bootstrap the core instance and the section of mobile-core.json that corresponds to that service.
+
+#### Writing a Service Module
+A service interface will be responsible for building requests, parsing responses, handling errors.  Services may depend on each other and the method of service composition is currently under evaluation. However, services will have a resolution order where a request is passed along a chain until it is ready to be executed.  Mobile Core will provide the services `threading`, `logging`, and `networking`.  ServiceModules will implement the service module class and provide a default no argument constructor.  
+
+Example KeyCloakService
```

Comment:
Keycloak service may not be the best use case here as it may be considered an wrapper for networking layer.
It will be cool to see sync service as we already have it extracted and we can simply import that into first production ready service

🗣  Thing to talk about.